packages/runtime/container-runtime/src/dataStoreContexts.ts

```diff
@@ -45,50 +49,84 @@ import { FluidDataStoreContext } from "./dataStoreContext";
     public get disposed() { return this.disposeOnce.evaluated;}
     public readonly dispose = () => this.disposeOnce.value;
 
+    public notBoundLength() {
+        return this.notBoundContexts.size;
+    }
+
+    public isNotBound(id: string) {
+        return this.notBoundContexts.has(id);
+    }
+
     public has(id: string) {
         return this._contexts.has(id);
     }
 
-    public setupNew(context: FluidDataStoreContext) {
+    public get(id: string): FluidDataStoreContext | undefined {
+        return this._contexts.get(id);
+    }
+
+    /**
+     * When a context of the given id is about to be bound/attached, call this to update internal tracking
+     */
+    public notifyOnBeforeBind(id: string) {
+        assert(this.notBoundContexts.has(id), "Store being bound should be in not bounded set");
+        this.notBoundContexts.delete(id);
+    }
+
+    /**
+     * Add the given context, marking it as to-be-bound
+     */
+    public addUnboundContext(context: LocalFluidDataStoreContext) {
```

Comment:
i don't this this api needs to specify context, as the class is all context releated